In [1]:
# Importar librerías
import duckdb

In [2]:
# Crear la conexión a la db Chinook
con = duckdb.connect()
con.execute("INSTALL sqlite; LOAD sqlite;")
con.execute("ATTACH 'Chinook_Sqlite.sqlite' AS chinook (TYPE sqlite);")

In [3]:
# Traemos la tabla "Invoices" mediante un dataframe
invoices = con.sql("""
    SELECT *
    FROM chinook.Invoice;
""").df()
invoices.head(5)

,InvoiceId,CustomerId,InvoiceDate,BillingAddress,BillingCity,BillingState,BillingCountry,BillingPostalCode,Total
0,1,2,2021-01-01,Theodor-Heuss-Straße 34,Stuttgart,NaN,Germany,70174,1.98
1,2,4,2021-01-02,Ullevålsveien 14,Oslo,NaN,Norway,0171,3.96
2,3,8,2021-01-03,Grétrystraat 63,Brussels,NaN,Belgium,1000,5.94
3,4,14,2021-01-06,8210 111 ST NW,Edmonton,AB,Canada,T6G 2C7,8.91
4,5,23,2021-01-11,69 Salem Street,Boston,MA,USA,2113,13.86


In [4]:
# Visualizar las características de la tabla
invoices.describe()

,InvoiceId,CustomerId,InvoiceDate,Total
count,412.000000,412.000000,412,412.000000
mean,206.500000,29.929612,2023-06-26 08:05:49.514563,5.651942
min,1.000000,1.000000,2021-01-01 00:00:00,0.990000
25%,103.750000,15.000000,2022-03-27 00:00:00,1.980000
50%,206.500000,30.000000,2023-06-22 12:00:00,3.960000
75%,309.250000,45.000000,2024-09-26 06:00:00,8.910000
max,412.000000,59.000000,2025-12-22 00:00:00,25.860000
std,119.078406,17.010585,NaN,4.745320


In [6]:
invoices.shape

(412, 9)

# **Ejercicio 4.1**

Carga la tabla Invoice como en el ejemplo y responde por escrito, sin buscar la respuesta "bonita":

**¿Cuántas filas y columnas tiene? ¿Qué representa cada fila?**

De acuerdo con **invoices.shape** nuestra tabla Invoice tiene 412 filas y 9 columnas (412,9)

¿Qué columna(s) parecen numéricas de verdad (tienen sentido sumarlas o promediarlas) y cuáles son números que en realidad son identificadores?

Considero que únicamente el campo "Total" es una columna numérica, con la cual podríamos hacer cálculos matemáticos. Luego los ID, aunque tienen valores numéricos, realmente no aplican para calculos sino más bien para establecer relaciones.

**Con .describe(): ¿algo en los mínimos, máximos o promedios te sorprende o no cuadra con lo que esperarías de una tienda de música?**

El total mínimo de compra es de 0.99, mientras que el máximo es de 25.86, con un promedio de 5.65. Esto nos podría indicar las compras de "discos" de música estás muy dispersos, idea que se recuerza con la desviación estándar 4.75 y el coeficiente de variación 4.75 / 5.65 = 84.07%.